### PR TITLE
Add GitHub workflow to bump formula versions

### PR DESCRIPTION
**Pull request by epk**
Signed-off-by: Aditya Sharma <git@adi.run>

### Description of your changes

Adds a GitHub workflow to bump formula versions

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

TODO after merging

### CODE REVIEW

File: .github/bump-formula.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    url = "https://cli.upbound.io/stable/#{NEW_VERSION}/bundle/#{formula}/#{arch}.tar.gz"
+    new_sha256 = `curl -sSL #{url} | sha256sum | cut -d' ' -f1`.strip
+
+    refs[arch] = {
+      old_sha256: old_sha256,
+      new_sha256: new_sha256
+    }
+  end
+
+  text = File.read("#{File.dirname(__FILE__)}/Formula/" + file)
```
Comment (hasheddan) [inline .github/bump-formula.rb:30]: Will this work in the .github directory?

Comment (epk) [reply .github/bump-formula.rb:30]: I think the action will execute from the root directory, so it should?

Comment (hasheddan) [reply .github/bump-formula.rb:30]: Ah nice 👍🏻 didn't realize action invoked using full relative path

**Review by hasheddan — APPROVED** [review]

File: .github/bump-formula.rb (hunk 1 above)
Comment (epk) [reply .github/bump-formula.rb:30]: You were right 🤦‍♂️ : https://github.com/upbound/homebrew-tap/pull/26

**Pull request merged**